README.md

```diff
@@ -1971,4 +1971,244 @@ You can create a `./gradlew checkstyle` command using the following:
 task checkstyle {
     dependsOn(checkstyleMain, checkstyleTest)
 }
-```
\ No newline at end of file
+```
+
+## TDD with Fibonacci
+
+As with all Test Driven Development, the key things to remember are:
+
+1. Red: Write as little code as possible to make your test fail.
+1. Green: Write as little code as possible to make your test pass.
+1. Refactor: Optimise your code for readability without changing its behaviour.
+
+First, create a test class that calls the intended target class:
+
+``` java
+public class FibonacciTest {
+    
+    Fibonacci target = new Fibonacci();
+
+}
+```
+
+Then create that target class.
+
+``` java
+public class Fibonacci {
+}
+```
+
+Second, add a simple method to the test class.
+
+``` java
+public class FibonacciTest {
+
+    public void shouldBe() {
+        Fibonacci target = new Fibonacci();
+        target.get(0);
+    }
+
+}
+```
+
+Then create the method `get` in the target class.
+
+``` java
+public class Fibonacci {
+
+    public int get() {
+        return 0;
+    }
+    
+}
+```
+
+Third, create a simple test in the test class.
+
+``` java
+import static org.junit.Assert.assertEquals;
+
+public class FibonacciTest {
+
+    @Test
+    public void shouldBe() {
+        Fibonacci target = new Fibonacci();
+        assertEquals(1,target.get());
+    }
+
+}
+```
+
+Then modify the method in the target class.
+
+``` java
+public class Fibonacci {
+
+    public int get() {
+        return 1;
+    }
+    
+}
+```
+
+Fourth, add more tests/assertions to the test class.
+
+``` java
+import static org.junit.Assert.assertEquals;
+
+public class FibonacciTest {
+
+    @Test
+    public void shouldBe() {
+        Fibonacci target = new Fibonacci();
+        assertEquals(1,target.get());
+        assertEquals(3,target.get());
+    }
+
+}
+```
+
+Then update the target class.
+
+``` java
+public class Fibonacci {
+
+    public int get(int n) {
+        if (n < 3) {
+            return 1;
+        }
+        return 2;
+    }
+}
+```
+
+There is now a lot of refactoring that can be done to the test class.
```

Comment:
woah, we jumped from 2 assertions to 5. That's not refactoring.

Reply:
Whoops!